src/main/java/mate/academy/bookshop/service/impl/BookServiceImpl.java

```diff
@@ -1,28 +1,42 @@
 package mate.academy.bookshop.service.impl;
 
 import java.util.List;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookshop.dto.BookDto;
+import mate.academy.bookshop.dto.CreateBookRequestDto;
+import mate.academy.bookshop.exception.EntityNotFoundException;
+import mate.academy.bookshop.mapper.BookMapper;
 import mate.academy.bookshop.model.Book;
 import mate.academy.bookshop.repository.impl.BookRepositoryImpl;
 import mate.academy.bookshop.service.BookService;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+@RequiredArgsConstructor
 @Service
 public class BookServiceImpl implements BookService {
-    private BookRepositoryImpl bookRepository;
+    private final BookRepositoryImpl bookRepository;
+    private final BookMapper bookMapper;
 
-    @Autowired
-    public BookServiceImpl(BookRepositoryImpl bookRepository) {
-        this.bookRepository = bookRepository;
+    @Override
+    public BookDto save(CreateBookRequestDto requestDto) {
+        Book book = bookMapper.toModel(requestDto);
+        return bookMapper.toDto(bookRepository.save(book));
     }
 
     @Override
-    public Book save(Book book) {
-        return bookRepository.save(book);
+    public BookDto findById(Long id) {
+        Optional<Book> bookById = bookRepository.findBookById(id);
+
+        return bookMapper.toDto(bookById.orElseThrow(
+                () -> new EntityNotFoundException("Can't find a book by ID: " + id)
+        ));

```

Comment:
You can use Optional.map for mapping it into DTO and inline byId variable. It would be prettier.

Reply:
Agree with @TarasYurkovskyi , it can be simplified :)